### ISSUE TITLE
Bug: Cannot resolve "@lexical/react/shared/useDecorators"

### ISSUE THREAD

**Issue by mr-josh** [facebook/lexical#4414]
When trying to make a custom plugin in react, I was using the PlainText plugin as reference and wanted to copy its usage of `useDecorators`. However when trying to import it via `import { useDecorators } from "@lexical/react/shared/useDecorators";`, vite throws the following error:

```
Internal server error: Failed to resolve import "@lexical/react/shared/useDecorators" from "src/components/testing/Lexical/Lexical.tsx". Does the file exist?
```

Looking into the node modules, it looks like only the type definitions have come across.

![image](https://user-images.githubusercontent.com/18021383/234483417-cdada1a0-50e9-429a-8b63-3ecd07ae856c.png)

I've been able to mitigate this for the most part by just [copying the source](https://github.com/facebook/lexical/blob/main/packages/lexical-react/src/shared/useDecorators.tsx) from the repository and having my own `useDecorators` file.

Lexical version: `lexical@0.10.0`
Lexical/react version: `@lexical/react@0.10.0`

## Steps To Reproduce

1. Install `@lexical/react`
2. Import `useDecorators` from `@lexical/react/shared/useDecorators`

Link to code example:
[![Edit competent-ptolemy-kd9kgk](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/competent-ptolemy-kd9kgk?fontsize=14&hidenavigation=1&theme=dark)

## The current behavior

Build tool fails to find and import `useDecorators`.

## The expected behavior

The expected hook is imported.

**Comment by acywatson** [comment]
The shared module is intentionally not exposed right now.

**Comment by thegreatercurve** [comment]
I don't think we necessarily want to expose these hooks. They're implementation details more than anything for the rich text and plain text plugins. I want us to be able to freely change them without any fear of the repercussions.
